packages/ui5-task-stringreplacer/lib/stringreplacer.js

```diff
@@ -0,0 +1,78 @@
+const replaceStream = require("replacestream");
+const dotenv = require("dotenv");
+const log = require("@ui5/logger").getLogger(
+  "builder:customtask:stringreplacer"
+);
+
+dotenv.config();
+
+// get all environment variables
+const envVariables = process.env;
+let placeholderStrings = [];
+
+// loop through env variables to find keys which are having prefix 'stringreplacer'
+if (typeof envVariables === "object") {
+  for (key in envVariables) {
+    // env variable should start with 'stringreplacer' and should in format 'stringreplacer.placeholder'
+    if (/^stringreplacer\.(.+)$/i.test(key)) {
+      let placeholderString = key.split(".")[key.split(".").length - 1];
```

Comment:
last comment: please use the regex to extract the placeholderstring - otherwise you only extract the last segment!

`let placeholderString = /^stringreplacer\.(.+)$/i.exec(key)[1];`

Reply:
I considered only last segment as placeholderstring till now. If considering entire path as placeholderstring is better, I will do this change.